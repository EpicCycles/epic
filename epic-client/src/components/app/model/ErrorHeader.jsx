import React, {Fragment} from "react";
import * as PropTypes from "prop-types";
import {fixedHeaderClassname, gridHeaderClass} from "./helpers/display";

const ErrorHeader = (props) => {
    return <div className={gridHeaderClass(props.className, 0, props.lockedColumn)}>
        Errors
    </div>;
};
ErrorHeader.defaultProps = {
    lockedColumn: false,
    className: "",
};
ErrorHeader.propTypes = {
    lockedColumn: PropTypes.bool,
    className: PropTypes.string,
};
export default ErrorHeader;