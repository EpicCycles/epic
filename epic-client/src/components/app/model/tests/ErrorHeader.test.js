import React from 'react';
import ErrorHeader from "../ErrorHeader";

test('should display error header', () => {
    const component = shallow(<ErrorHeader/>);
    expect(component).toMatchSnapshot();
});
test('should display error header with locking and an extra class', () => {
    const component = shallow(<ErrorHeader lockedColumn={true} className={"pink"}/>);
    expect(component).toMatchSnapshot();
});