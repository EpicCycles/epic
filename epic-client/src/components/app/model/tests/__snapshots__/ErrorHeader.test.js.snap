// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display error header 1`] = `
<div
  className=" grid-item--header "
>
  Errors
</div>
`;

exports[`should display error header with locking and an extra class 1`] = `
<div
  className="pink grid-item--header grid-header--fixed-left"
>
  Errors
</div>
`;
